src/providers/ModalProvider/Modal/index.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+  memo,
+  MouseEvent,
+  ReactElement,
+  useCallback,
+  useEffect,
+} from 'react';
+import { useModal } from '..';
+
+import { ModalBackground, ModalContainer, ModalContent } from './index.styled';
+
+interface IProps {
+  id: number,
+  show: boolean,
+  children: ReactElement<any>,
+};
+
+const Modal = ({ id, show, children }: IProps) => {
+  const { deleteModal, closeModal } = useModal();
+
+  useEffect(() => {
+    if (!show) {
+      const timer = setTimeout(() => {
+        deleteModal(id);
+      }, 300);
```

Comment:
300 seems like a magic number here, it is not obvious why do we have it. Maybe it is worth to add a comment explaining that it is animation time. Another option would be to create a constant with appropriate name.